### PR TITLE
Make description of Samplers Locale independent

### PR DESCRIPTION
**Pull request by lmonkiewicz**
Fixes: #4880

### CODE REVIEW

File: sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/RateLimitingSampler.java [hunk 1]
```diff
@@ -62,11 +64,19 @@ public SamplingResult shouldSample(
 
   @Override
   public String getDescription() {
-    return String.format("RateLimitingSampler{%.2f}", maxTracesPerSecond);
+    return String.format("RateLimitingSampler{%s}", decimalFormat(maxTracesPerSecond));
```
Comment (jack-berg) [inline sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/RateLimitingSampler.java:67]: #nit: now that we're not relying on any decimal formatting, let's switch to string concatenation, which is generally favored over `String.format` in this repository: `"RateLimitingSampler{" + decimalFormat(maxTracesPerSecond) + "}"`

Same advice applies in `TraceIdRatioBasedSampler`.

**Review by jack-berg — COMMENTED** [review]
Just one small comment. Thanks!

**Comment by lmonkiewicz** [comment]
> Just one small comment. Thanks!

Changed.

**Review by jack-berg — APPROVED** [review]

**Review by trask — COMMENTED** [review]
@jack-berg what do you think about calculating it once in the constructor? `getDescription` is not supposed to be called a lot, but just in case...

and I think the extra small amount of memory is ok since there shouldn't be a huge number of Sampler instances

**Review by trask — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

**Pull request merged**